sdk/core/core-util/test/public/reshape.spec.ts

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { describe, it, assert, expect } from "vitest";
+import { reshape } from "../../src/reshape";
+
+describe("reshape", function () {
```

Comment:
Not sure this could be a concern but I am a little bit worried about the performance. The selection and applying would require extra calculations and also every time for `reshape` we would produce extra objects.

Reply:
Got it, if we do this in place then there would be no extra objects.